gtsfm/densify/mvs_utils.py

```diff
@@ -199,3 +216,65 @@ def downsample_point_cloud(
 
     points_downsampled, rgb_downsampled = open3d_vis_utils.convert_colored_open3d_point_cloud_to_numpy(pcd)
     return points_downsampled, rgb_downsampled
+
+
+def compute_downsampling_psnr(original_point_cloud: np.ndarray, downsampled_point_cloud: np.ndarray) -> float:
+    """Compute PSNR between original point cloud and downsampled point cloud
+    A larger PSNR shows smaller distances between:
+        1. the nearest neighbors of a point in the original point cloud found in the downsampled point cloud
+        1. the nearest neighbors of a point in the downsampled point cloud found in the original point cloud
+
+    Ref: Schnabel, R., & Klein, R. (2006, July). Octree-based Point-Cloud Compression. In PBG@SIGGRAPH (pp. 111-120).
+        https://diglib.eg.org/xmlui/bitstream/handle/10.2312/SPBG.SPBG06.111-120/111-120.pdf?sequence=1
+
+    Args:
+        original_point_cloud: original dense point cloud before downsampling, in shape of (N, 3)
+        downsampled_point_cloud: dense point cloud after downsampling, in shape of (N', 3), where N' <= N
+
+    Returns:
+        float: PSNR between original point cloud and downsampled point cloud
```

Comment:
nit: "float representing PSNR between original point cloud and downsampled point cloud"

Reply:
marking as resolved, thanks!